backend/src/apiserver/server/api_util.go

```diff
@@ -114,11 +116,32 @@ func getJobIdFromResourceReferencesV1(resourceRefs []*apiv1beta1.ResourceReferen
 	return jobId
 }
 
-// Converts structpb.Struct to a yaml string.
-func protobufStructToYamlString(s *structpb.Struct) (string, error) {
-	bytes, err := yaml.Marshal(s)
-	if err != nil {
-		return "", util.Wrap(err, "Failed to convert a protobuf struct to a yaml string")
+// Converts pipeline spec in the form of structpb.Struct to a yaml string.
+// Handles three possible cases:
+// 1) pipeline spec passed directly
+// 2) pipeline spec along with platform specific configs, each as a sub struct
+// 3) pipeline spec as a sub struct, with platform specific field empty
+func pipelineSpecStructToYamlString(s *structpb.Struct) (string, error) {
```

Comment:
Could you please add unit-tests to check the behavior for these three cases and some expected errors (nils, maybe)?

Reply:
This is a really good point! I just added more tests.